encoding/wkb/sql_example_test.go

```diff
@@ -74,3 +74,98 @@ func Example_value() {
 	// Output:
 	// 1 rows affected
 }
+
+func Example_scan_different_shapes() {
+	type Shape struct {
+		Name string
+		Geom wkb.Geom
+	}
+
+	db, mock, err := sqlmock.New()
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer db.Close()
+
+	mock.ExpectQuery(`SELECT name, ST_AsBinary\(geom\) FROM shapes`).
+		WillReturnRows(
+			sqlmock.NewRows([]string{"name", "location"}).
+				AddRow("Point", geomtest.MustHexDecode("0101000000000000000000F03F0000000000000040")).
+				AddRow("LineString", geomtest.MustHexDecode("010200000002000000000000000000F03F000000000000004000000000000008400000000000001040")).
+				AddRow("Polygon", geomtest.MustHexDecode("01030000000100000004000000000000000000F03F00000000000000400000000000000840000000000000104000000000000014400000000000001840000000000000F03F0000000000000040")),
+		)
+
+	rows, err := db.Query(`SELECT name, ST_AsBinary(geom) FROM shapes`)
+	if err != nil {
+		log.Fatal(err)
+	}
+	for rows.Next() {
+		var s Shape
+		err := rows.Scan(&s.Name, &s.Geom)
+		if err != nil {
+			log.Fatal(err)
+		}
+		fmt.Printf("%s: %v\n", s.Name, s.Geom.FlatCoords())
+	}
+
+	// Output:
+	// Point: [1 2]
+	// LineString: [1 2 3 4]
+	// Polygon: [1 2 3 4 5 6 1 2]
+}
+
+func Example_value_different_shapes() {
```

Comment:
Same comments removing `\`s from queries and printing `Coords` instead of `FlatCoords` apply.